sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -1190,4 +1201,141 @@ public HttpPipeline getHttpPipeline() {
+    }
+
+    Mono<Response<EventRoute>> getEventRouteWithResponse(String eventRouteId, Context context)
+    {
+        return this.protocolLayer.getEventRoutes().getByIdWithResponseAsync(eventRouteId, context);
+    }
+
+    /**
+     * Delete an event route.
+     * @param eventRouteId The Id of the event route to delete.
+     * @return An empty mono.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteEventRoute(String eventRouteId)
+    {
+        return deleteEventRouteWithResponse(eventRouteId)
+            .flatMap(voidResponse -> Mono.empty());
+    }
+
+    /**
+     * Delete an event route.
+     * @param eventRouteId The Id of the event route to delete.
+     * @return A {@link Response} containing an empty mono.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> deleteEventRouteWithResponse(String eventRouteId)
+    {
+        return withContext(context -> deleteEventRouteWithResponse(eventRouteId, context));
+    }
+
+    Mono<Response<Void>> deleteEventRouteWithResponse(String eventRouteId, Context context)
+    {
+        return this.protocolLayer.getEventRoutes().deleteWithResponseAsync(eventRouteId, context);
+    }
+
+    /**
+     * List all the event routes that exist in your digital twins instance.
+     * @return A {@link PagedFlux} that contains all the event routes that exist in your digital twins instance.
+     *         This PagedFlux may take multiple service requests to iterate over all event routes.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<EventRoute> listEventRoutes()
+    {
+        return listEventRoutes(new EventRoutesListOptions());
+    }
+
+    /**
+     * List all the event routes that exist in your digital twins instance.
+     * @param options The optional parameters to use when listing event routes. See {@link EventRoutesListOptions} for more details
+     *                on what optional parameters can be set.
+     * @return A {@link PagedFlux} that contains all the event routes that exist in your digital twins instance.
+     *         This PagedFlux may take multiple service requests to iterate over all event routes.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<EventRoute> listEventRoutes(EventRoutesListOptions options)
+    {
+        return new PagedFlux<>(
+            () -> withContext(context -> listEventRoutesFirstPage(options, context)),
+            nextLink -> withContext(context -> listEventRoutesNextPage(nextLink, context)));
+    }
+
+    PagedFlux<EventRoute> listEventRoutes(EventRoutesListOptions options, Context context)
+    {
+        return new PagedFlux<>(
+            () -> listEventRoutesFirstPage(options, context),
+            nextLink -> listEventRoutesNextPage(nextLink, context));
+    }
+
+    Mono<PagedResponse<EventRoute>> listEventRoutesFirstPage(EventRoutesListOptions options, Context context) {
```

Comment:
nit: this can be inline in line 1234, right?

Reply:
This function and the next page function are called in two places, so I'd rather keep them in their own function